go/tools/gopackagesdriver/gopackagesdriver_test.go

```diff
@@ -259,7 +260,117 @@ func TestExternalTests(t *testing.T) {
+	wd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// format filepaths for overlay request
+	helloPath := path.Join(wd, "hello.go")
+	subhelloPath := path.Join(wd, "subhello/subhello.go")
+
+	expectedImportsPerFile := map[string][]string{
+		helloPath: []string{"fmt"},
+		subhelloPath: []string{"os", "encoding/json"},
+	}
+
+	overlayDriverRequest := map[string]map[string]string{
+		"overlay": {
+			helloPath:    `
+				package hello
+				import "fmt"
+				import "unknown/unknown-package"
+				func main() {
+					invalid code
+
+				}`,
+			subhelloPath: `
+				package subhello
+				import "os"
+				import "encoding/json"
+				func main() {
+					fmt.Fprintln(os.Stderr, "Subdirectory Hello World!")
+				}
+			`,
+		},
+	}
+
+	// encode the overlay content
+	for key, value := range overlayDriverRequest["overlay"] {
+		encodedValue := base64.StdEncoding.EncodeToString([]byte(value))
+		overlayDriverRequest["overlay"][key] = encodedValue
+	}
+
+	var buf bytes.Buffer
+	encoder := json.NewEncoder(&buf)
+	if err := encoder.Encode(overlayDriverRequest); err != nil {
+		t.Errorf("hello package not found in response root")
+	}
+	overlayDriverRequestStr := buf.String()
+
+	// run the driver with the overlay
+	helloResp := runForTest(t, overlayDriverRequestStr, ".", "file=hello.go")
+	subhelloResp := runForTest(t, overlayDriverRequestStr, "subhello", "file=subhello.go")
+
+	// get root packages
+	var helloPkg, subhelloPkg *FlatPackage
+	for _, p := range helloResp.Packages {
+		if p.ID == helloResp.Roots[0] {
+			helloPkg = p
+		}
+	}
+	for _, p := range subhelloResp.Packages {
+		if p.ID == subhelloResp.Roots[0] {
+			subhelloPkg = p
+		}
+	}
+	if helloPkg == nil {
+		t.Errorf("hello package not found in response root")
+	}
+	if subhelloPkg == nil {
+		t.Errorf("subhello package not found in response")
+	}
+
+	for _, expectedImport := range expectedImportsPerFile[helloPath] {
+		found := false
+		for importPath, _ := range helloPkg.Imports {
+			if importPath == expectedImport {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Expected hello import %s not found in parsed imports: %v", expectedImport, helloPkg.Imports)
+		}
+	}
+
+	for _, expectedImport := range expectedImportsPerFile[subhelloPath] {
+		found := false
+		for importPath, _ := range subhelloPkg.Imports {
+			if importPath == expectedImport {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Expected subhello import %s not found in parsed imports: %v", expectedImport, helloPkg.Imports)
+		}
+	}
+
+	if len(helloPkg.Imports) != len(expectedImportsPerFile[helloPath]) {
+		t.Errorf("hello imports have unexpected length")
+	}
+
+	if len(subhelloPkg.Imports) != len(expectedImportsPerFile[subhelloPath]) {
+		t.Errorf("subhello imports have unexpected length")
+	}
+}
+
+
+func runForTest(t *testing.T, DriverRequestJson string, relativeWorkingDir string, args ...string) driverResponse {
```

Comment:
Why don't we take in the DriverRequest struct here and JSON encode it from here? This seems to be a better location for it that in the individual test cases.

Reply:
Also variable naming should be `driverRequestJson`

Reply:
fixed